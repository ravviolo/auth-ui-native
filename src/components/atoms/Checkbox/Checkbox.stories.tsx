import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { StoryScreen } from 'storybook/StoryScreen';

import { Checkbox } from './Checkbox';

storiesOf('Checkbox', module)
  .addDecorator((getStory) => <StoryScreen center>{getStory()}</StoryScreen>)
  .add('Default', () => <Checkbox testID={text('testID', 'test-id')} />);
