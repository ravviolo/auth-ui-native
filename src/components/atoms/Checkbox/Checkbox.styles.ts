import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  checkbox: {
    borderRadius: 4,
    borderWidth: 1,
  },
});
