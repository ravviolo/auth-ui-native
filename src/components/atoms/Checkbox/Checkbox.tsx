import ExpoCheckbox from 'expo-checkbox';
import { forwardRef, useImperativeHandle, useState } from 'react';
import { StyleSheet } from 'react-native';
import { colors } from 'theme';

export interface CheckboxHandle {
  getValue: () => boolean;
  clear: () => void;
}

export interface Props {
  testID: string;
}

export const Checkbox = forwardRef<CheckboxHandle, Props>(({ testID }, ref) => {
  const [isChecked, setIsChecked] = useState(false);

  useImperativeHandle(ref, () => ({
    getValue: () => isChecked,
    clear: () => {
      setIsChecked(false);
    },
  }));

  return (
    <ExpoCheckbox
      color={isChecked ? colors.primary : colors.textLight}
      style={styles.checkbox}
      testID={testID}
      value={isChecked}
      onValueChange={setIsChecked}
    />
  );
});

const styles = StyleSheet.create({
  checkbox: {
    borderRadius: 4,
    borderWidth: 1,
  },
});
