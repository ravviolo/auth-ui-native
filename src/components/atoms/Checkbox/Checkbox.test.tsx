import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { Checkbox } from './Checkbox';

import type { Props } from './Checkbox';

describe('Checkbox', () => {
  const props: Props = {
    testID: 'checkbox-test-id',
  };
  it('should initially render unchecked', () => {
    render(<Checkbox {...props} />);

    const checkbox = screen.getByTestId('checkbox-test-id');

    expect(checkbox.props.accessibilityState).toHaveProperty('checked', false);
  });

  it('should toggle checked state', () => {
    render(<Checkbox {...props} />);

    const checkbox = screen.getByTestId('checkbox-test-id');

    fireEvent.press(checkbox);
    expect(checkbox.props.accessibilityState).toHaveProperty('checked', true);

    fireEvent.press(checkbox);
    expect(checkbox.props.accessibilityState).toHaveProperty('checked', false);
  });

  it('should match snapshot', () => {
    const tree = testRenderer.create(<Checkbox {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
