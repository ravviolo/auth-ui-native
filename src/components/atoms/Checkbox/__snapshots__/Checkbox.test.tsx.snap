// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match snapshot 1`] = `
<View
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "borderColor": "#657786",
        "borderRadius": 2,
        "borderWidth": 2,
        "height": 20,
        "width": 20,
      },
      Object {
        "borderRadius": 4,
        "borderWidth": 1,
      },
      false,
      Object {
        "backgroundColor": undefined,
        "borderColor": "#9E9E9E",
      },
      undefined,
      false,
    ]
  }
  testID="checkbox-test-id"
/>
`;
